### PR TITLE
[Azure] [Container_instance] Add dimensions and metric_type metadata for the container_instance data_stream

### PR DESCRIPTION
**Pull request by tetianakravchenko**
## What does this PR do?

- remove duplicated fields
- instead of using wildcard in the field name - use predefined field names
  - for azure.dimensions - dimension name is predefined - https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-containerinstance-containergroups-metrics
  - list of metrics - https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-containerinstance-containergroups-metrics, default aggregation is `avg`
- used fields for dimensions:
  - `cloud.region`
  - `azure.resource.id` - this field includes `azure.subscription_id`, `azure.resource.group`, `azure.resource.name`, format:
  ```
  /subscriptions/{azure.subscription_id}/resourceGroups/{azure.resource.group}/providers/Microsoft.ContainerInstance/containerGroups/{azure.resource.name}
  ```
  -  `azure.dimensions.container_name` - this field is not available for 2 metrics: `network_bytes_transmitted_per_second.avg` and `network_bytes_received_per_second.avg`
  -  `azure.timegrain`
- metrics - use predefined metrics - since in the configuration is defined list of metrics

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by tetianakravchenko** [comment]
@zmoog I couldn't find those fields - https://github.com/elastic/integrations/blob/main/packages/azure_metrics/data_stream/container_instance/fields/fields.yml#L1

could you please check if this field is not added by mistake?

**Comment by zmoog** [comment]
> could you please check if this field is not added by mistake?

Yep, I'll check it out!

File: packages/azure_metrics/changelog.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+- version: "1.0.17"
+  changes:
+    - description: Add dimension and metric_type metadata to the container_instance data_stream
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/5123
```
Comment (agithomas) [inline packages/azure_metrics/changelog.yml:5]: Please correct the PR link.

File: packages/azure_metrics/changelog.yml [hunk 2]
```diff
@@ -1,3 +1,8 @@
+- version: "1.0.17"
+  changes:
+    - description: Add dimension and metric_type metadata to the container_instance data_stream
```
Comment (agithomas) [inline packages/azure_metrics/changelog.yml:3]:
```suggestion
    - description: Add dimension and metric_type metadata to the container_instance datastream
```

File: packages/azure_metrics/data_stream/container_instance/fields/package-fields.yml [hunk 3]
```diff
@@ -3,6 +3,7 @@
   description: ""
   fields:
     - name: timegrain
+      dimension: true
```
Comment (agithomas) [inline packages/azure_metrics/data_stream/container_instance/fields/package-fields.yml:6]: Recommendation: Please include an inline comment stating the reason for adding `timegrain` as a dimension field as it is not an obvious choice.

File: packages/azure_metrics/manifest.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 name: azure_metrics
 title: Azure Resource Metrics
-version: 1.0.16
+version: 1.0.17
```
Comment (agithomas) [inline packages/azure_metrics/manifest.yml:3]: Why not `1.1.0` ? Any specific reason ?

File: packages/azure_metrics/changelog.yml (hunk 1 above)
Comment (tetianakravchenko) [reply packages/azure_metrics/changelog.yml]: done - https://github.com/elastic/integrations/pull/7127/commits/7a77aa7b36497b758794cee0304e5d1991895792

File: packages/azure_metrics/data_stream/container_instance/fields/package-fields.yml [hunk 5]
```diff
@@ -25,6 +26,7 @@
 
         - name: id
           type: keyword
+          dimension: true
```
[the remark this answers is not in the document]
Comment (tetianakravchenko) [reply packages/azure_metrics/data_stream/container_instance/fields/package-fields.yml:32]: added - https://github.com/elastic/integrations/pull/7127/commits/7a77aa7b36497b758794cee0304e5d1991895792

File: packages/azure_metrics/data_stream/container_instance/fields/package-fields.yml (hunk 3 above)
Comment (tetianakravchenko) [reply packages/azure_metrics/data_stream/container_instance/fields/package-fields.yml:7]: done - https://github.com/elastic/integrations/pull/7127/commits/7a77aa7b36497b758794cee0304e5d1991895792

File: packages/azure_metrics/manifest.yml (hunk 4 above)
Comment (tetianakravchenko) [reply packages/azure_metrics/manifest.yml:3]: no specific reason, I did it similarly to the AWS package, also I've noticed that previous changes in this package were also added with `1.0.X`

**Review by agithomas — APPROVED** [review]
LGTM!

File: packages/azure_metrics/data_stream/container_instance/fields/package-fields.yml [hunk 6]
```diff
@@ -55,17 +60,33 @@
...
-        Metrics returned.
+    - name: dimensions
+      type: group
+      fields:
+        - name: container_name
+          type: keyword
+          dimension: true
+          description: The container name
 
+    - name: metrics
```
Comment (tetianakravchenko) [inline packages/azure_metrics/data_stream/container_instance/fields/package-fields.yml:71]: as discussed with @zmoog : remove `metrics` field, move those metrics under `container_instance`

**Comment by mlunadia** [comment]
@bturquet can you help us getting these reviewed? We have a long queue of Azure packages waiting on it. Merci

File: packages/azure_metrics/manifest.yml (hunk 4 above)
Comment (zmoog) [reply packages/azure_metrics/manifest.yml:3]: Enabling TSDB seems worth a bump to the minor version, but I have no strong opinions. I'll leave the decision to you, TSDB folks.

**Review by zmoog — APPROVED** [review]

File: packages/azure_metrics/manifest.yml (hunk 4 above)
Comment (tetianakravchenko) [reply packages/azure_metrics/manifest.yml:3]: @zmoog there will be another PR to enable TSDB by default (similar to this [PR](https://github.com/elastic/integrations/pull/7030) and additionally adding stack restriction), this PR just adds metadata to the fields, that will be used by TSDB

File: packages/azure_metrics/data_stream/container_instance/fields/package-fields.yml (hunk 3 above)
Comment (tetianakravchenko) [reply packages/azure_metrics/data_stream/container_instance/fields/package-fields.yml:7]: FYI - there was created an [issue](https://github.com/elastic/integrations/issues/7162) to clarify meaning of `timegrain` field

**Pull request merged**